src/public/modules/forms/admin/views/edit-fields.client.modal.html

```diff
@@ -1087,6 +1087,85 @@ <h2 class="modal-title">
         </div>
         <!-- End custom validation field -->
 
+        <!-- Short Text -->
+
+        <div
+          ng-show="vm.field.fieldType === 'textfield'"
+          class="edit-fields-line"
+        ></div>
+
+        <div ng-if="vm.field.fieldType === 'textfield'" class="row">
+          <div class="row"><br /></div>
+          <div class="toggle-option">
+            <div class="col-xs-8 label-custom label-medium">
+              Enable pre-fill
+              <i
+                class="glyphicon glyphicon-question-sign"
+                uib-tooltip="Allow this field to be pre-populated via URL"
+                tooltip-trigger="'click mouseenter'"
+              >
+              </i>
+            </div>
+            <div class="col-xs-4 field-input">
+              <label
+                class="toggle-selector pull-right"
+                ng-class="vm.field.allowPrefill ? 'toggle-selector-on' : ''"
+                onclick=""
+              >
+                <input type="checkbox" ng-model="vm.field.allowPrefill" />
+                <div class="toggle-selector-switch">
+                  <i
+                    ng-class="vm.field.allowPrefill ? 'bx bx-check' : 'bx bx-x' "
+                  ></i>
+                </div>
+              </label>
+            </div>
+          </div>
+        </div>
+
+        <div
+          ng-if="vm.field.fieldType === 'textfield' && vm.field.allowPrefill"
+          class="row"
+        >
+          <div class="row"><br /></div>
+          <div class="col-md-12 label-custom label-medium label-bottom">
+            Field ID
+          </div>
+          <div class="col-md-12">
+            <input
+              type="text"
+              ng-disabled="true"
+              class="input-custom input-medium"
+              value="{{vm.field._id || 'Field ID will be generated after field is saved'}}"
+            />
+            <i
+              class="bx bx-copy bx-md copy-icon-overlay prefill-copy-icon"
+              ng-if="vm.field._id"
+              ngclipboard
+              ng-click="vm.fieldIdCopied()"
+              data-clipboard-text="{{vm.field._id}}"
+            ></i>
```

Comment:
i'm looking at the code example from the [docs](http://sachinchoolur.github.io/ngclipboard/) - should we be using `data-clipboard-action="copy"` and `data-clipboard-target`, as well as using `ngclipboard-success` (and `ngclipboard-error` instead of `ng-click`?

```html
<!-- Target -->
<textarea id="bar">Mussum ipsum cacilds...</textarea>

<!-- Trigger -->
<button class="btn" ngclipboard data-clipboard-action="cut" data-clipboard-target="#bar">
    Cut to clipboard
</button>
```

Reply:
edited, although this means that the fieldId field has to be set to readonly instead of disabled, slight change from original design (see [link](https://github.com/zenorocha/clipboard.js/issues/41), the issue seems to affect chrome now)

Reply:
in that case, we should invest in the CSS to have it work with `readonly`. generally, we shouldn't be overriding designers' design without consulting with them first.

Reply:
edited